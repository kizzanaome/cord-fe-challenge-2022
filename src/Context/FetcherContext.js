import React, { Component } from 'react';
import * as fetcher from "../fetcher";



const FetcherContext = React.createContext();


export class FetcherProvider extends Component {

    state = {
        genreOptions: [],
    };


    componentDidMount() {
        this.loadAllGenres();
    }

    
    /**
    * LISTS ALL GENRES
    */
    loadAllGenres = async () => {
        const server_response = await fetcher.getAllGenres();
        if (server_response.status === 200) {
            this.setState({
                genreOptions: server_response.data.genres,
            })
        } else {
            this.setState({
                genreOptions: []
            })
        }
    }


    render() {

        return (
            <FetcherContext.Provider value={{
                genreOptions: this.state.genreOptions
            }}>
                {this.props.children}
            </FetcherContext.Provider>
        );
    }

}


export default FetcherContext;
