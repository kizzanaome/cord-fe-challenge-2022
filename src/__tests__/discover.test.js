import React from 'react';
import MobileHeader from '../pages/discover/index';
import MobilePageTitle from '../pages/discover/index';
import FetcherContext, { FetcherProvider } from '../Context/FetcherContext'
import Discover from '../pages/discover'




import { render, screen, clean } from '@testing-library/react';

import renderer from 'react-test-renderer'


describe("renders the discover UI correctly", () => {


    beforeEach(() => {



        // const TestComponet = () => {
        //     const genreOptions = React.useContext(FetcherContext)
        //     return (
        //         <FetcherProvider value={{ genreOptions }}>
        //             <Discover />
        //         </FetcherProvider>
        //     );
        // }

    });


    it("renders mobile header correctly", () => {
        const TestComponet = () => {
            React.useContext(FetcherContext)
            const tree = renderer.create(
                <MobileHeader>
                    <MobilePageTitle>Discover</MobilePageTitle>
                </MobileHeader>
            ).toJSON();
            expect(tree).toMatchSnapshot();

        }

    })

    it("renders mobile title correctly", () => {

        const TestComponet = () => {
            const genreOptions = React.useContext(FetcherContext)

            const component = <MobilePageTitle>Discover</MobilePageTitle>
            const tree = renderer.create(component).toJSON();
            expect(tree).toMatchSnapshot();
            // );
        }

    })


})
