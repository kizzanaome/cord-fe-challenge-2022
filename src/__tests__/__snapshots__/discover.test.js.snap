// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the discover UI correctly renders  correctly 1`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders correctly 1`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders mobile header correctly 1`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders mobile header correctly 2`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders mobile headers correctly 1`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders mobile title correctly 1`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;

exports[`renders the discover UI correctly renders mobile title correctly 2`] = `
<main
  className="sc-bjUoiL eehIho"
>
  <div
    className="sc-hHLeRK dXYShW"
  >
    <h1
      className="sc-dmRaPn jDetwL"
    >
      Discover
    </h1>
  </div>
  <div
    className="sc-kgflAQ eUbBEa"
  >
    <div
      className="sc-jSMfEi htNodg search_bar_wrapper"
    >
      <img
        alt="Magnifying glass"
        htmlFor="keyword_search_input"
        src="search-icon-yellow.png"
        width="25"
      />
      <input
        id="keyword_search_input"
        onChange={[Function]}
        placeholder="Search for movies"
        type="text"
      />
    </div>
    <div
      className="filterIcon"
    >
      <img
        alt="menu"
        src="filter-icon.png"
      />
    </div>
  </div>
  <strong
    className="sc-fLlhyt Sikxz"
  >
    0
     movies
  </strong>
  <div
    className="sc-dIouRR hYLHJF"
  >
    <div
      className="sc-gKXOVf czoAbi"
    >
      <div
        className="sc-iBkjds kHPdLD search_inputs_cont"
      >
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Magnifying glass"
            htmlFor="keyword_search_input"
            src="search-icon-yellow.png"
            width="25"
          />
          <input
            id="keyword_search_input"
            onChange={[Function]}
            placeholder="Search for movies"
            type="text"
          />
        </div>
        <div
          className="sc-jSMfEi htNodg search_bar_wrapper"
        >
          <img
            alt="Calendar icon"
            htmlFor="year_search_input"
            src="year-icon.png"
            width="25"
          />
          <input
            id="year_search_input"
            onChange={[Function]}
            placeholder="Year of release"
            type="number"
          />
        </div>
      </div>
      <div
        className="sc-iBkjds drTrpK"
      >
        <h3
          className="sc-ftvSup hgbAvM"
        >
          Movies
        </h3>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select genre(s)
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select min. vote
            </div>
          </div>
        </div>
        <div
          className="sc-bczRLJ"
        >
          <div
            className="sc-gsnTZi cJhKTw"
            onClick={[Function]}
          >
            <div
              className="sc-dkzDqf cvHdPD"
            >
              +
            </div>
            <div>
              Select language
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-idiyUo ehectm"
  >
    <p>
      
    </p>
    <p>
      Loading...
    </p>
    <div
      className="sc-fEOsli lgBRRH"
    />
  </div>
</main>
`;
